streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -213,6 +213,36 @@ trait ZStreamPlatformSpecificConstructors { self: ZStream.type =>
       .managed(is)
       .flatMap(fromInputStream(_, chunkSize))
 
+  /**
+   * Creates a stream from a callback that writes to [[java.io.OutputStream]].
+   * Note: the input stream will be closed after the `write` is done.
+   */
+  def fromOutputStream(
```

Comment:
Most methods that capture effects in ZIO are called `effect<something>`. Should we call this `effectOutputStream`? Or maybe `fromOutputStreamWriter`?

Reply:
I think I like `fromOutputStreamWriter` more (because of the symmetry with `fromInputStream`) - but I doubt we will be able to come up with a good name in general. So as long as it has `OutputStream` in it (for auto completion) I am fine.

Reply:
`fromOutputStreamWriter` it is!